logstash-core/src/test/java/org/logstash/jackson/StreamReadConstraintsUtilTest.java

```diff
@@ -0,0 +1,205 @@
+
+            assertThatThrownBy(configuredUtil::validateIsGlobalDefault).isInstanceOf(IllegalStateException.class).hasMessageContaining(MAX_NESTING_DEPTH.getPropertyName());
+
+            configuredUtil.applyAsGlobalDefault();
+            assertThatCode(configuredUtil::validateIsGlobalDefault).doesNotThrowAnyException();
+        });
+    }
+
+    @Test
+    public void configuresMaxNestingDepthInvalid() {
+        final Properties properties = new Properties();
+        properties.setProperty(MAX_NESTING_DEPTH.getPropertyName(), "NaN");
+
+        fromProperties(properties, (configuredUtil, defaults) -> {
+            assertThatThrownBy(configuredUtil::get)
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessageContaining(MAX_NESTING_DEPTH.getPropertyName());
+        });
+    }
+
+    @Test
+    public void configuresMaxNestingDepthNegative() {
+        final Properties properties = new Properties();
+        properties.setProperty(MAX_NESTING_DEPTH.getPropertyName(), "-1000");
+
+        fromProperties(properties, (configuredUtil, defaults) -> {
+            assertThatThrownBy(configuredUtil::get)
+                    .isInstanceOf(IllegalArgumentException.class)
+                    .hasMessageContaining(MAX_NESTING_DEPTH.getPropertyName());
+        });
+    }
+
+    @Test
+    public void validatesApplication() {
+        final Properties properties = new Properties();
+        properties.setProperty(StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option1", "100");
+        properties.setProperty(StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option2", "100");
+        properties.setProperty(MAX_NESTING_DEPTH.getPropertyName(), "1010");
+        properties.setProperty(MAX_STRING_LENGTH.getPropertyName(), "1011");
+        properties.setProperty(MAX_NUMBER_LENGTH.getPropertyName(), "1110");
+
+        fromProperties(properties, (configuredUtil, defaults) -> {
+            configuredUtil.applyAsGlobalDefault();
+            configuredUtil.validateIsGlobalDefault();
+        });
+
+        assertLogObserved(Level.INFO, "override `" + MAX_NESTING_DEPTH.getPropertyName() + "` configured to `1010`");
+        assertLogObserved(Level.INFO, "override `" + MAX_STRING_LENGTH.getPropertyName() + "` configured to `1011`");
+        assertLogObserved(Level.INFO, "override `" + MAX_NUMBER_LENGTH.getPropertyName() + "` configured to `1110`");
+
+        assertLogObserved(Level.WARN, "override `" + StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option1` is unknown and has been ignored");
+        assertLogObserved(Level.WARN, "override `" + StreamReadConstraintsUtil.PROP_PREFIX + "unsupported-option1` is unknown and has been ignored");
+    }
+
+    private void assertLogObserved(final Level level, final String... messageFragments) {
+        assertThat(loggingSpyResource.getLogEvents())
+                .withFailMessage("Expected an {}-level log event containing {}", level, messageFragments)
```

Comment:
The [failed test output in CI](https://buildkite.com/elastic/logstash-pull-request-pipeline/builds/1891#01935560-e7bd-4918-8dc7-8d6ccf0d95aa)
```console
org.logstash.jackson.StreamReadConstraintsUtilTest > validatesApplication FAILED
--
  | java.lang.AssertionError: Expected an {}-level log event containing {}
  | at org.logstash.jackson.StreamReadConstraintsUtilTest.assertLogObserved(StreamReadConstraintsUtilTest.java:193)
  | at org.logstash.jackson.StreamReadConstraintsUtilTest.validatesApplication(StreamReadConstraintsUtilTest.java:181)


```
suggests interpolation is not quite setup right. The [AssertJ](https://assertj.github.io/doc/#assertj-core-overriding-error-message) docs have an example with `%s` style which seems to fix these up locally. 
```diff
diff --git a/logstash-core/src/test/java/org/logstash/jackson/StreamReadConstraintsUtilTest.java b/logstash-core/src/test/java/org/logstash/jackson/StreamReadConstraintsUtilTest.java
index 076521a16..9ace4616a 100644
--- a/logstash-core/src/test/java/org/logstash/jackson/StreamReadConstraintsUtilTest.java
+++ b/logstash-core/src/test/java/org/logstash/jackson/StreamReadConstraintsUtilTest.java
@@ -6,6 +6,7 @@ import org.junit.Rule;
 import org.junit.Test;
 import org.logstash.log.LoggingSpyResource;

+import java.util.Arrays;
 import java.util.Properties;
 import java.util.function.BiConsumer;
 import org.apache.logging.log4j.Level;
@@ -188,7 +189,7 @@ public class StreamReadConstraintsUtilTest {

     private void assertLogObserved(final Level level, final String... messageFragments) {
         assertThat(loggingSpyResource.getLogEvents())
-                .withFailMessage("Expected an {}-level log event containing {}", level, messageFragments)
+                .withFailMessage("Expected an %s-level log event containing %s", level, Arrays.toString(messageFragments))
                 .filteredOn(logEvent -> logEvent.getLevel().equals(level))
                 .anySatisfy(logEvent -> assertThat(logEvent.getMessage().getFormattedMessage()).contains(messageFragments));
     }
```
One thing that i find curious is that when I run the tests in this class in isolation (`./gradlew :logstash-core:javaTests --tests org.logstash.jackson.StreamReadConstraintsUtilTest`) The error is not reproduced. It only appears when running the whole suite. Its not immediately clear to me as to why.